protocols/v2/subprotocols/common-messages/src/setup_connection.rs

```diff
@@ -84,8 +84,23 @@ impl<'decoder> SetupConnection<'decoder> {
 
                 work_selection && version_rolling
             }
-            // TODO
-            _ => todo!(),
+            Protocol::JobDeclarationProtocol => {
+                let available = available_flags.reverse_bits();
+                let required = required_flags.reverse_bits();
+
+                let requires_async_job_mining_passed = (required >> 31) & 1 > 0;
+                let requires_async_job_mining_self = (available >> 31) & 1 > 0;
+
+                let specific_flags_check =
+                    !requires_async_job_mining_self || requires_async_job_mining_passed;
```

Comment:
seems that this produces false when the async mining job is available in the upstream, but it is not required by the downstream. In my opinion this should produce true, not false. 
Also, this produces true when async mining job is required by the downstrem bu not available by the upstream. I think that this should produce false.
Passed  $  Self  $  !Self || Passed
      T       $    F    $            T
      T       $    T    $            T
      F       $    T    $            F
      F       $    F    $            T

Reply:
@lorbax has a point, here's a refined truth table:
| available | required | check |
|-----------|----------|-------|
| T         | T        | T     |
| T         | F        | T     |
| F         | T        | F    |
| F         | F        | T     |

I'm also writing some suggestions on the code